### PR TITLE
filter: fix caseSensitive not work on match

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve? 
caseSensitive not work on Match 

### What is changed and how it works?
add ToLower if !caseSensitive

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Review by kennytm — COMMENTED** [review]
please add a test case?

File: pkg/filter/filter_test.go [hunk 1]
```diff
@@ -187,6 +187,9 @@ func (s *testFilterSuite) TestCaseSensitive(c *C) {
 	expected := []*Table{{"foo2", "b"}, {"Foo4", "dfoo"}, {"5", "5"}}
 	c.Logf("got %+v, expected %+v", actual, expected)
 	c.Assert(actual, DeepEquals, expected)
+
+	inputTable := &Table{"FOO", "a"}
```
Comment (lichunzhu) [inline pkg/filter/filter_test.go:191]: How about adding a positive case? for example, `&Table{"foo3", "c"}

Comment (3pointer) [reply pkg/filter/filter_test.go:191]: sure, add one

**Review by lichunzhu — COMMENTED** [review]
LGTM

**Review by kennytm — APPROVED** [review]
LGTM

**Pull request merged**